timemachine/maps/terminal_bonds.py

```diff
@@ -0,0 +1,236 @@
+"""Summarize HarmonicBonds as Intervals in R+, construct maps between Intervals, & apply these maps to bond lengths"""
+
+from dataclasses import dataclass
+from typing import Tuple
+
+import numpy as np
+from jax import config, jacobian, jit
+from jax import numpy as jnp
+from jax import vmap
+from numpy.typing import NDArray as Array
+
+config.update("jax_enable_x64", True)
+
+from typing import List
+
+import networkx as nx
+
+from timemachine.constants import BOLTZ, DEFAULT_TEMP
+
+
+@dataclass
+class Interval:
+    lower: float
+    upper: float
+
+    @property
+    def width(self) -> float:
+        return self.upper - self.lower
+
+    def __post_init__(self):
+        assert self.width > 0
+        assert self.lower > 0
+
+
+@dataclass
+class Gaussian:
+    mean: float
+    stddev: float
+
+    def to_interval(self, sigma_thresh=20) -> Interval:
+        r = self.stddev * sigma_thresh
+        interval = Interval(self.mean - r, self.mean + r)
+        return interval
+
+    @classmethod
+    def from_harmonic_bond(cls, force_constant, eq_length, temperature=DEFAULT_TEMP):
+        return cls(eq_length, np.sqrt(BOLTZ * temperature / force_constant))
+
+
+@jit
+def interval_map(x, src_lb, src_ub, dst_lb, dst_ub):
+    """map x in Interval[src_lb, src_ub] to y in Interval[dst_lb, dst_ub]"""
+
+    scale_factor = (dst_ub - dst_lb) / (src_ub - src_lb)
+
+    in_support = (x >= src_lb) & (x <= src_ub)
+    mapped = dst_lb + (x - src_lb) * scale_factor
+
+    return jnp.where(in_support, mapped, np.nan)
+
+
+def conf_map(x, bond, param) -> Tuple[Array, Array]:
+    """Apply map to a single bond in a conformer x, return (updated x, logdetjac)"""
+    a, b = bond
+    dim = 3
+
+    def f(xy, param):
```

Comment:
nit: change f to a more descriptive name

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/801/commits/c3c8d0a4f15211de051e53e64fb8f86ea9037a01